05week/bankAccount.js

```diff
@@ -0,0 +1,96 @@
+'use strict'
+
+// Account Class
+// attributes / fields:
+// - account number // this is passed in to the constructor
+// - owner //this is passed in to the constructor
+// - a list of transactions  // starts of empty
+// methods:
+// - balance(): loop through all the transactions, 
+//   and return the current balance
+//-  addTransaction(payee, amount); // checks to see if the amount is valid (ie cannot debit more than the balance)
+//
+//-  instead of addTransaction():
+//   deposit(amt);
+//   charge(payee, amount)
+//
+
+class BankAccount {
+  constructor(accountNumber, owner) {
+    this.accountNumber = accountNumber;
+    this.owner = owner;
+    this.transactions = [];
+  }
+
+  balance() {
+    if (this.transactions.length > 0) {
+      let currentBalance = 0;
+      for (let i=0; i<this.transactions.length; i++) {
+        currentBalance += this.transactions[i].amount;
+      }
+      return currentBalance;
+    } else {
+      return `There are ${this.transactions.length} transactions to list.`
+    }
+  }
+
+  deposit(amt) {
+    if (amt < 0) {
+      console.log(`Can't deposit a negative number.`);
+      return `Please enter a valid deposit`
```

Comment:
returning a string does n't make sense for a method like this, since the users of these methods are not end users, but other developers.  it would make more sense if this method returned true/false depending if the deposit was accepted, or threw an exception if the input was not valid.